api/sessiontokenloginprovider.go

```diff
@@ -54,6 +54,16 @@ type sessionTokenLoginProvider struct {
 	updateAccountDetailsFunc func(string) error
 }
 
+// Token implements the [LoginProvider.Token] method.
+// Returns the session token obtained after performing the login flow.
+// Returns an ErrorTokenNotAvailable error if no token available.
+func (p *sessionTokenLoginProvider) Token() (string, error) {
+	if p.sessionToken == "" {
+		return "", ErrorTokenNotAvailable
+	}
```

Comment:
@tlm Here is the other use of the new error types. I'm still basing the result off of the default type for a string but at least now the logic is encapsulated within the provider rather than leaking out into the rest of the code.
I could extend this to include a check for whether the existing token is still valid and return an error if it's not. Lmk what you think about that.